lib/bulk/common.js

```diff
@@ -1130,9 +1134,9 @@ class BulkOperationBase {
       delete finalOptions.bypassDocumentValidation;
     }
 
-    // Set an operationIf if provided
-    if (this.operationId) {
-      config.resultHandler.operationId = this.operationId;
+    // Set an operationId if provided
+    if (this['operationId']) {
+      config.resultHandler['operationId'] = this['operationId'];
```

Comment:
The operationId property doesn't exist on this class it exists on the subclasses, I can do this to make the type error go away or add `this.operationId = undefined` in the constructor.

The other places I did it were for example, options objects where we add some property on to the options before sending it deeper into the driver, since the prop hadn't existed until that point the prop didn't exist on the type of the object, I could just call the object `any` but there was some typing already in place that I didn't want to lose. Future work that I suppose I should make a ticket for would be to reduce `any`'s and `obj['blah']` as part of increasing our typing strength

Reply:
Hmm, I think I'd prefer just having a `// @ts-ignore` on those lines. In actual TS we'd probably use an abstract class for something like this in order to get good checks, but I'd rather add comments than tweak the code in cases where we're not actually getting any benefit from the checks.

Reply:
I'm inclined to agree, it also marks the issues for future fixing with a consistent comment, cus we also use `obj['prop']` notation elsewhere on purpose